tests/test_experiment.py

```
@@ -0,0 +1,173 @@
+from pathlib import Path
+
+SRC = Path(__file__).resolve().parents[1] / "src"
+sys.path.insert(0, str(SRC))
+
+import signing  # noqa: E402  (torch-free)
+from artifacts import build_merkle_manifest  # noqa: E402  (torch-free)
+
+try:
+    import torch  # noqa: F401
+    HAS_TORCH = True
+except Exception:
```

Ruff (0.15.17)

[warning] Do not catch blind exception: `Exception`

(BLE001)